library/lmots.c

```diff
@@ -181,12 +182,17 @@ static int create_digit_array_with_checksum( const mbedtls_lmots_parameters_t *p
     if( status != PSA_SUCCESS )
         goto exit;
 
-    status = psa_hash_finish( &op, out,
-                              MBEDTLS_LMOTS_N_HASH_LEN(params->type),
+    status = psa_hash_finish( &op, temp,
+                              MBEDTLS_LMOTS_N_HASH_LEN_MAX,
                               &output_hash_len );
     if( status != PSA_SUCCESS )
         goto exit;
 
+    memcpy( out, temp, MBEDTLS_LMOTS_N_HASH_LEN( params->type ) );
```

Comment:
If SHA256/192 isn't made a distinct PSA hash algorithm, the hash-and-truncate dance should be made an auxiliary function.

Reply:
Where do you see this function be placed? Within the same file or somewhere more reusable?

Reply:
in `lmots.c`, and declared only in `library/*.h`, I think, since this is (so far) specific to the LMS/XMMS family.